src/main/java/graphql/execution/defer/DeferSupport.java

```diff
@@ -22,12 +26,15 @@
 
     private final AtomicBoolean deferDetected = new AtomicBoolean(false);
     private final Deque<DeferredCall> deferredCalls = new ConcurrentLinkedDeque<>();
-    private final SingleSubscriberPublisher<ExecutionResult> publisher = new SingleSubscriberPublisher<>();
+    private final SingleSubscriberPublisher<DeferredExecutionResult> publisher = new SingleSubscriberPublisher<>();
+    private final ValuesResolver valuesResolver = new ValuesResolver();
 
-    public boolean checkForDeferDirective(List<Field> currentField) {
+    public boolean checkForDeferDirective(List<Field> currentField, Map<String, Object> variables) {
         for (Field field : currentField) {
-            if (field.getDirective(Directives.DeferDirective.getName()) != null) {
-                return true;
+            Directive directive = field.getDirective(Directives.DeferDirective.getName());
+            if (directive != null) {
+                Map<String, Object> argumentValues = valuesResolver.getArgumentValues(Directives.DeferDirective.getArguments(), directive.getArguments(), variables);
+                return (Boolean) argumentValues.get("if");
```

Comment:
can this be null? => unboxing will cause NPE

Reply:
no - its default is true

Reply:
now made it non null as an arg declaration